include/nodejs/node_osrm_support.hpp

```diff
@@ -249,6 +256,9 @@ inline engine_config_ptr argumentsToEngineConfig(const Nan::FunctionCallbackInfo
         engine_config->max_results_nearest = static_cast<int>(max_results_nearest->NumberValue());
     if (max_alternatives->IsNumber())
         engine_config->max_alternatives = static_cast<int>(max_alternatives->NumberValue());
+    if (max_radius_map_matching->IsNumber())
+        engine_config->max_radius_map_matching =
+            static_cast<boost::optional<double>>(max_radius_map_matching->NumberValue());
```

Comment:
Not sure the cast is legal.. I think you want to use `boost::make_optional` here (or change to plain double as mentioned above).